### PR TITLE
[5.4] Postgres: Grammar for pivot-related DELETE 

### PR DESCRIPTION
**Pull request by ddzobov**
Added grammar for DELETE with USES clause

### CODE REVIEW

**Comment by ddzobov** [comment]
#19989

File: src/Illuminate/Database/Query/Grammars/PostgresGrammar.php [hunk 1]
```diff
@@ -227,6 +227,55 @@ public function prepareBindingsForUpdate(array $bindings, array $values)
...
+     * Compile a delete query that uses joins.
+     *
+     * @param  \Illuminate\Database\Query\Builder  $query
+     * @param  string  $table
+     * @param  array  $where
+     * @return string
+     */
+    protected function compileDeleteWithJoins($query, $table)
+    {
+        $using = ' USING '.collect($query->joins)->map(function ($join) use ($query, &$using_tables) {
```
Comment (ddzobov) [inline src/Illuminate/Database/Query/Grammars/PostgresGrammar.php]: thank you, removed old part of code

File: src/Illuminate/Database/Query/Grammars/PostgresGrammar.php [hunk 2]
```diff
@@ -266,7 +266,7 @@ protected function compileDeleteWithoutJoins($query, $table)
      */
     protected function compileDeleteWithJoins($query, $table)
     {
-        $using = ' USING '.collect($query->joins)->map(function ($join) use ($query, &$using_tables) {
+        $using = ' USING '.collect($query->joins)->map(function ($join) use ($query) {
```
Comment (srmklive) [inline src/Illuminate/Database/Query/Grammars/PostgresGrammar.php]: I don't think you need to have `use($query)` as well here.

**Comment by tillkruss** [comment]
Nice, thanks!

**Pull request merged**